api/v1alpha1/limitador_types.go

```diff
@@ -102,6 +107,93 @@ type LimitadorList struct {
 	Items           []Limitador `json:"items"`
 }
 
+// StorageType defines the valid options for storage
+// +kubebuilder:validation:Enum=memory;redis;redis_cached
+type StorageType string
+
+const (
+	StorageTypeInMemory    StorageType = "memory"
+	StorageTypeRedis                   = "redis"
+	StorageTypeRedisCached             = "redis_cached"
+)
+
+// Storage contains the options for Limitador counters database or in-memory data storage
+type Storage struct {
+	// +optional
+	Redis *Redis `json:"redis,omitempty"`
+
+	// +optional
+	RedisCached *RedisCached `json:"redis-cached,omitempty"`
+}
+
+func (s *Storage) Validate() bool {
+	return s.Redis != nil && s.Redis.ConfigSecretRef != nil ||
+		s.RedisCached != nil && s.RedisCached.ConfigSecretRef != nil
+}
+
+func (s *Storage) SecretRef() *corev1.ObjectReference {
+	if s.Redis != nil {
+		return s.Redis.ConfigSecretRef
+	}
+	return s.RedisCached.ConfigSecretRef
+}
+
+func (s *Storage) Config(url string) []string {
+	if s.Redis != nil {
+		return []string{
+			StorageTypeRedis,
+			url,
+		}
+	} else if s.RedisCached != nil {
+		return []string{
+			StorageTypeRedisCached,
+			url,
+			fmt.Sprintf("--ttl %d", s.RedisCached.Options.TTL),
```

Comment:
they can be nil, add them only if they exist. Which is good because defaults will be defined at the limitador level and not defined in the operator.

Reply:
Actually, they cannot be nil because you added the kubebuilder defaults. 

I would rather, remove defaults defined in the operator (i.e. the kubebuilder tags)  and add to the CLI params only if they specified in the CR (i.e. the field is not nil)